test/h5test.c

```diff
@@ -2236,3 +2240,160 @@ h5_local_srand(unsigned int seed)
 {
     next_g = seed;
 }
+
+#ifdef H5_HAVE_ROS3_VFD
+
+/*
+ * Load AWS credentials from ~/.aws/config and ~/.aws/credentials
+ */
+herr_t
+h5_load_aws_profile(const char *profile_name, bool *profile_found, char *key_id_out, size_t key_id_out_len,
```

Comment:
This function replaces `H5FD__s3comms_load_aws_profile()` / `H5FD__s3comms_load_aws_creds_from_file()` from the libcurl backend and delegates loading of credentials for testing to the aws-c-s3 library.